src/test/java/com/ibm/cloud/cloudant/kafka/connect/TombstoneTest.java

```diff
@@ -0,0 +1,100 @@
+import static org.easymock.EasyMock.replay;
+
+public class TombstoneTest {
+
+    private static final String connectionName = "_mock";
+
+    @Test
+    public void testDeletedDocumentEmitsTombstones() throws Exception{
+
+        String id = "123";
+
+        Document document = new Document();
+        document.setId(id);
+        document.setDeleted(Boolean.TRUE);
+
+        ChangesResultItem changesResultItem = PowerMock.createMock(ChangesResultItem.class);
+        Change change = PowerMock.createMock(Change.class);
+        Cloudant mockCloudant = PowerMock.createMock(Cloudant.class);
+        Response<ChangesResult> mockResponse = PowerMock.createMock(Response.class);
+        ChangesResult mockResult = PowerMock.createMock(ChangesResult.class);
+        expect(mockResponse.getResult()).andReturn(mockResult);
+        expect(mockResult.getResults()).andReturn(Collections.singletonList(changesResultItem));
+        expect(mockResult.getResults()).andReturn(Collections.singletonList(changesResultItem));
+        expect(mockResult.getLastSeq()).andReturn("100");
+        expect(changesResultItem.getChanges()).andReturn(Collections.singletonList(change));
+        expect(changesResultItem.getDoc()).andReturn(document);
+        expect(changesResultItem.getId()).andReturn(id);
+        expect(mockCloudant.postChanges(anyObject())).andReturn(new ServiceCall<ChangesResult>() {
+            @Override
+            public ServiceCall<ChangesResult> addHeader(String s, String s1) {
+                return null;
+            }
+
+            @Override
+            public Response<ChangesResult> execute() throws RuntimeException {
+                return mockResponse;
+            }
+
+            @Override
+            public void enqueue(ServiceCallback<ChangesResult> serviceCallback) {
+
+            }
+
+            @Override
+            public Single<Response<ChangesResult>> reactiveRequest() {
+                return null;
+            }
+
+            @Override
+            public void cancel() {
+
+            }
+        });
+        ClientManagerUtils.addClientToCache(connectionName, mockCloudant);
+        CloudantSourceTask cloudantSourceTask = new CloudantSourceTask();
+        Map<String, String> configMap = new HashMap<>();
+        configMap.put("name", connectionName);
+        configMap.put("cloudant.since", "1000");
+        configMap.put("cloudant.url", "http://foo");
+        configMap.put("cloudant.db", "foo");
+        configMap.put("topics", "foo");
+        replay(mockCloudant);
+        replay(mockResponse);
+        replay(mockResult);
+        replay(change);
+        replay(changesResultItem);
+        cloudantSourceTask.start(configMap);
+        List<SourceRecord> records = cloudantSourceTask.poll();
+        Assert.assertEquals(2, records.size());
+        Assert.assertNotNull(records.get(0).value());
+        Assert.assertNull(records.get(1).value());
+        // TODO unpack record json and assert on it
```

Comment:
I don't think this TODO is necessary, asserting the record value is `null` is sufficient to know it is a tombstone.

Arguably we could have a second test that validates that we _don't get a second record_ when the change wasn't a deleted doc, but I expect assertions elsewhere are covering that.